.github/actions/export-ocm-fragments/action.yaml

```diff
@@ -161,34 +170,82 @@ runs:
 
         pprint.pprint(ocm_artefacts)
 
-        ocm_artefacts_yaml_bytes = yaml.safe_dump(ocm_artefacts).encode('utf-8')
-        ocm_artefacts_digest = hashlib.sha1(ocm_artefacts_yaml_bytes).hexdigest()
-        ocm_artefacts_fname = f'{ocm_artefacts_digest}.ocm-artefacts'
-        if ctx := '${{ inputs.ctx }}':
-          ocm_artefacts_fname = f'{ctx}-{ocm_artefacts_fname}'
-
         ocm_tar = tarfile.open(
-          name=tempfile.NamedTemporaryFile(delete=False).name,
+          name=tempfile.NamedTemporaryFile(
+            delete=False,
+            dir='.', # do not use /tmp to avoid unnecessary copy after finalised
+          ).name,
           mode='w:gz',
         )
 
-        ocm_artefacts_info = tarfile.TarInfo(name=ocm_artefacts_fname)
-        ocm_artefacts_info.size = len(ocm_artefacts_yaml_bytes)
-        buf = io.BytesIO(ocm_artefacts_yaml_bytes)
-        ocm_tar.addfile(ocm_artefacts_info, buf)
-
+        orig_name_to_digest_names = dict()
         # add blobs
         if os.path.isdir(blobs_dir := '${{ inputs.blobs-directory }}'):
           for fname in os.listdir(blobs_dir):
             fpath = os.path.join(blobs_dir, fname)
             if not os.path.isfile(fpath):
               continue
 
+            fhash = hashlib.sha256()
+            with open(fpath, 'rb') as f:
+              while chunk := f.read(4096):
+                fhash.update(chunk)
+
+            # symlink, unless file already is named according to its content-hash-digest
+            fhashname = f'sha256:{fhash.hexdigest()}'
+            if fhash != fhashname:
+              os.symlink(fname, fhashname)
+              orig_name_to_digest_names[fname] = fhashname
+              ocm_tar.add(
+                name=fhashname,
+                arcname=f'blobs.d/{fhashname}',
+              )
+
             ocm_tar.add(
               name=fpath,
               arcname=f'blobs.d/{fname}',
             )
 
+        # patch local-access if referencing local-files using non-digest-filename
+        for artefact in sources + resources:
+          if not (access := artefact.get('access', None)):
+            continue
+          if ocm.AccessType(access.get('type')) is not ocm.AccessType.LOCAL_BLOB:
+            continue
+          if not (local_ref := access.get('localReference', None)):
+            continue
+          if local_ref.startswith('sha256:'):
+            continue
+          if not os.path.exists(local_fpath := os.path.join(blobs_dir, local_ref)):
+            continue
+          # local file exists, and it does not meet content-addresssing scheme (
+          # <alg>:<hexdigest>). We should find it from previous step where we calculated
+          # digests, and created symlinks.
+          hashname = orig_name_to_digest_names.get(local_ref, None)
+          if not hashname:
+            print(f'WARNING: did not find local blobfile for {artefact=}')
```

Comment:
Shouldn't we rather raise?

Reply:
I think no. Rationale: this action is intended to be fairly low-level. Callers might decide to already upload blobs into target-oci-artefact (where ocm-component-descriptor will be uploaded). As an alternative, missing files might also be added at a later point-in-time (theoretically).

Hence, I did on purpose not raise here. We might consider adding a "strict" mode, though (where we would consider this an error).